### ISSUE TITLE
For Class-based syntax on Orchestrators / Activities that have no return value, how should we type them?

### ISSUE THREAD

**Issue by Tyler-V** [microsoft/durabletask-dotnet#75]
In our solution, we have orchestrators that have an input and no output... and we also have activities that have an input and sometimes no output

Using Class-based syntax requires a TInput, TOutput on the Base and in the activities not returning a value compiles fine but for Orchestrators a value must be returned

Our Orchestrators call sub orchestrators, downstream (4 total) with fan-in / fan-out on the activities

What is the best way to resolve this?

For example,

```
[DurableTask(nameof(FileUploadOrchestrator))]
public class FileUploadOrchestrator : TaskOrchestratorBase<FileUploadOrchestratorInput, CaptureRequestedOrchestratorInput>
{
    protected override async Task<CaptureRequestedOrchestratorInput> OnRunAsync(TaskOrchestrationContext taskOrchestrationContext,
        FileUploadOrchestratorInput input)
    {
        foreach (var blobMetadata in input.Blobs)
        {
            var captureRequestedOrchestratorInput =
                await taskOrchestrationContext.CallActivityAsync<CaptureRequestedOrchestratorInput>(nameof(FileUploadActivity), blobMetadata);

            await taskOrchestrationContext.CallSubOrchestratorAsync(nameof(CaptureRequestedOrchestrator), input: captureRequestedOrchestratorInput);
        }

        return null; //Nothing to return but required for compilation
    }
}
```

**Comment by jviau** [comment]
For now, the best way is to have just `object` and return `null`. Alternatively, you can define a class like `Empty` or `Void` and return a singleton of that, to indicate to your callers this return value is meaningless.

**Comment by Tyler-V** [comment]
That makes sense @jviau thanks!

After serialization is fixed in related #36 we are excited to try this out again
